src/btree/bt_curprev.c

```diff
@@ -541,6 +543,16 @@ __cursor_row_prev(
         if ((ins = cbt->ins) != NULL) {
             key->data = WT_INSERT_KEY(ins);
             key->size = WT_INSERT_KEY_SIZE(ins);
+            /*
+             * If the cursor has prefix search configured we can early exit here if the key we are
+             * visiting is before our prefix.
+             */
+            if (prefix_used && __wt_prefix_match(prefix, key) > 0) {
+                /* It is not okay for the user to have a custom collator. */
```

Comment:
Given that we check the custom collator when we do the cursor reconfigure I think we can remove this assert and the comment in all the locations we have it in bt_curprev and bt_curnext.